titus-server-federation/src/main/java/com/netflix/titus/federation/service/AggregatingJobServiceGateway.java

```diff
@@ -135,15 +138,24 @@ public AggregatingJobServiceGateway(GrpcConfiguration grpcConfiguration,
         }
         JobManagementServiceStub client = optionalClient.get();
 
-        JobDescriptor withStackName = addStackName(jobDescriptor);
+        JobDescriptor enrichedJobDescriptor;
+        if (federationConfiguration.isFederationJobIdCreationEnabled()) {
+            String federatedJobId = UUID.randomUUID().toString();
+            enrichedJobDescriptor = addJobAttributes(jobDescriptor,
+                    ImmutableMap.of(JOB_ATTRIBUTES_STACK, federationConfiguration.getStack(),
+                            JOB_ATTRIBUTES_FEDERATED_JOB_ID, federatedJobId));
```

Comment:
It would be good to also have the "routed cell" from line 128 included in the attributes as well.

Reply:
No.  `addStackName` is adding the stack from here:
```
federationConfiguration.getStack()
```
which is different than the cell the job is being routed to.

Reply:
ah, you are right! We fill in cell info later down the road (`JobAttributes.JOB_ATTRIBUTES_CELL`)